tests/unit/src/util/process.ts

```diff
@@ -0,0 +1,230 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import { stub, SinonStub } from 'sinon';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import * as processUtil from 'src/util/process';
+import { LogStream } from 'src/log';
+
+let module: any;
+let execStub: SinonStub;
+let spawnStub: SinonStub;
+
+registerSuite({
+	name: 'util/process',
+
+	before() {
+		execStub = stub();
+		spawnStub = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		module = loadModule('src/util/process', {
+			child_process: {
+				exec: execStub,
+				spawn: spawnStub
+			}
+		});
+	},
+
+	afterEach() {
+		execStub.reset();
+		spawnStub.reset();
+	},
+
+	promisify: (() => {
+		let proc: any;
+
+		return {
+			beforeEach() {
+				proc = {
+					stdout: { pipe: stub() },
+					stderr: { pipe: stub() },
+					on: stub()
+				};
+
+				stub(processUtil, 'exec').returns(proc);
+			},
+
+			afterEach() {
+				(<SinonStub> processUtil.exec).restore();
+			},
+
+			async 'eventually resolves the returned promise'() {
+				const promise = processUtil.promisify(processUtil.exec('test'));
+
+				proc.on.lastCall.args[1](0);
+				assert.equal(proc, await promise);
+
+				return promise;
+			},
+
+			async 'child process exits with code other than 0; eventually rejects the returned promise'() {
+				let promise;
+
+				try {
+					promise = processUtil.promisify(processUtil.exec('test'));
+
+					proc.on.lastCall.args[1](1);
+					assert.equal(proc, await promise);
```

Comment:
The await should throw an error before the assert equal has a chance to run... I think this is leftover copy/paste code from the previous test?

Can we clean this up so if the promise doesn't reject the test will fail